Keas.Mvc/ClientApp/components/Keys/AssignKey.tsx

```diff
@@ -45,10 +45,11 @@ export default class AssignKey extends React.Component<IProps, IState> {
   constructor(props) {
     super(props);
     this.state = {
-      date: moment().add(3, "y"),
-      error: "",
+      date: (!!this.props.selectedKey && !!this.props.selectedKey.assignment) 
```

Comment:
Let's think about adding some helpers in the future that can determine if an assignment exists and to set the default expiration date.  Something like if `hasAssignment(this.props.selectedKey) { }` and `getDefaultExpiration()`, etc, etc.